provider/icu4x-datagen/src/main.rs

```diff
@@ -588,6 +580,74 @@ fn main() -> eyre::Result<()> {
     Ok(())
 }
 
+macro_rules! cb {
+    ($($marker:path = $path:literal,)+ #[experimental] $($emarker:path = $epath:literal,)+) => {
+        fn all_markers() -> Vec<DataMarkerInfo> {
+            vec![
+                $(
+                    <$marker>::INFO,
+                )+
+                $(
+                    #[cfg(feature = "experimental")]
+                    <$emarker>::INFO,
+                )+
+            ]
+        }
+
+        fn marker_lookup() -> &'static HashMap<&'static str, Option<DataMarkerInfo>> {
+            use std::sync::OnceLock;
+            static LOOKUP: OnceLock<HashMap<&'static str, Option<DataMarkerInfo>>> = OnceLock::new();
+            LOOKUP.get_or_init(|| {
+                [
+                    ("core/helloworld@1", Some(icu_provider::hello_world::HelloWorldV1Marker::INFO)),
+                    (stringify!(icu_provider::hello_world::HelloWorldV1Marker).split("::").last().unwrap().trim(), Some(icu_provider::hello_world::HelloWorldV1Marker::INFO)),
+                    $(
+                        ($path, Some(<$marker>::INFO)),
+                        (stringify!($marker).split("::").last().unwrap().trim(), Some(<$marker>::INFO)),
+                    )+
+                    $(
+                        #[cfg(feature = "experimental")]
+                        ($epath, Some(<$emarker>::INFO)),
+                        #[cfg(feature = "experimental")]
+                        (stringify!($emarker).split("::").last().unwrap().trim(), Some(<$emarker>::INFO)),
+                        #[cfg(not(feature = "experimental"))]
+                        ($epath, None),
+                        #[cfg(not(feature = "experimental"))]
+                        (stringify!($emarker).split("::").last().unwrap().trim(), None),
+                    )+
+
+                ]
+                .into_iter()
+                .collect()
+            })
+        }
+
+        #[test]
+        fn test_lookup() {
+            assert_eq!(marker_lookup().get("AndListV1Marker"), Some(&Some(icu::list::provider::AndListV1Marker::INFO)));
+            assert_eq!(marker_lookup().get("list/and@1"), Some(&Some(icu::list::provider::AndListV1Marker::INFO)));
+            assert_eq!(marker_lookup().get("foo"), None);
+        }
```

Comment:
Nit: This test need not be inside the macro?

Reply:
I think it's nice to have it next to the function it tests

Reply:
I prefer if you move it 15 lines down and out of the macro